### PR TITLE
 Introduce facility to debug the DataRelayer

### PR DESCRIPTION
**Pull request by ktf**
As we develop the ability to customise the dispatching of computations
via the CompletionPolicy it became clear that a proper way of debugging
the state of the DataRelayer is fundamental to actually understand what
is going on with the system.

This introduces a simple GUI element which displays for each device the
state of the DataRelayer.

### CODE REVIEW

**Comment by ktf** [comment]
Obligatory screenshot:

image

**Comment by ktf** [comment]
There might be some bug fixes still, but already this is functional enough to watch processing behavior and discover issues. Since it's mostly GUI changes I am merging it. The `clang-format` remaining issue is bogus.

**Comment by dberzano** [comment]
Just to understand, the issue related to the representation of `timePipeline` is still there, right?

**Comment by ktf** [comment]
yes

File: Framework/Core/src/FrameworkGUIDebugger.cxx [hunk 1]
```diff
@@ -100,6 +104,117 @@ ImVec4 colorForLogLevel(LogParsingHelpers::LogLevel logLevel)
...
+{
+  if (mainStr.find(toMatch) == 0) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+void displayDataRelayer(DeviceMetricsInfo const& metrics,
+                        DeviceInfo const& info)
```
Comment (matthiasrichter) [inline Framework/Core/src/FrameworkGUIDebugger.cxx]: get a compiler warning here
```
/home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/src/FrameworkGUIDebugger.cxx:183:23: warning: declaration of ‘const o2::framework::MetricInfo& info’ shadows a parameter [-Wshadow]
     MetricInfo const& info = metrics.metrics[viewIndex.indexes[idx]];
                       ^~~~
/home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/src/FrameworkGUIDebugger.cxx:160:43: note: shadowed declaration is here
                         DeviceInfo const& info)
                                           ^~~~
```
I thought warnings are treated as errors in the CI, has this policy been policy changed? @dberzano , @sawenzel

File: Framework/Core/src/FrameworkGUIDebugger.cxx [hunk 2]
```diff
@@ -100,6 +104,117 @@ ImVec4 colorForLogLevel(LogParsingHelpers::LogLevel logLevel)
...
+    if (viewIndex.isComplete()) {
+      return viewIndex.h;
+    }
+    return 0;
+  };
+  auto getItem = [&metrics, &viewIndex](int const& record, size_t i) {
+    // Calculate the index in the viewIndex.
+    auto idx = record * viewIndex.h + i;
+    assert(viewIndex.indexes.size() > idx);
+    MetricInfo const& info = metrics.metrics[viewIndex.indexes[idx]];
```
Comment (matthiasrichter) [inline Framework/Core/src/FrameworkGUIDebugger.cxx]: this is the line where the parameter is shadowed, no problem though

**Pull request merged**